### PR TITLE
30 Day notice:Deprecation of FMUv1 and PX4IOv1 - Merge on Sept. 10th

### PR DESCRIPTION
**Pull request by davids5**
This PR removes the support for FMUv1 and PX4IOv1 HW from PX4. 

The ardrone_interface is also removed as it was only used by FMUv1.

See https://github.com/PX4/Firmware/issues/7758 

@hamishwillee FYI

### CODE REVIEW

**Comment by davids5** [comment]
@PX4TestFlights - Once CI completes, please flight test on V3, V4, V4Pro, V5, areofc - there should be no notable differences from master.

File: ROMFS/px4fmu_common/init.d/rcS [hunk 1]
```diff
@@ -343,46 +343,44 @@ then
 		if [ -f /etc/extras/px4io-v2.bin ]
 		then
 			set IO_FILE /etc/extras/px4io-v2.bin
```
Comment (davids5) [inline ROMFS/px4fmu_common/init.d/rcS]: @LorenzMeier  - pleas have a look at this. section. The assumption is that if the "/etc/extras/px4io-v2.bin" does not exist the PX4IO is not started. This is in contrast to using the "/etc/extras/px4io-v1.bin" that might not have existed and failing on each ${IO_FILE} access.

**Comment by santiago3dr** [comment]
flight with PixRacer (v4):
https://logs.px4.io/plot_app?log=51f5f51f-7e12-40b3-9e9b-187ae9b5dca3

compared to master under same conditions:
https://review.px4.io/plot_app?log=33175788-b701-49b2-878e-f10b533c11d7

good flight; no notable difference from current master 6e35cb0

**Comment by davids5** [comment]
@santiago3dr  - Thank you.

I expected V4 to be just fine - but I am glad you confirmed it.

V3 (mini, V2-No errata) , V4Pro, and aerofc will provide better test coverage for these changes.

**Comment by davids5** [comment]
Jenkins test this please.

**Comment by santiago3dr** [comment]
flight with pixhawk 1 (v2 no errata):
https://logs.px4.io/plot_app?log=ad6c92db-87d4-4c3e-833d-e92711ab3718

compared to master under same conditions:
https://logs.px4.io/plot_app?log=96a58323-02ce-4d29-aac6-bd96df1780f9

good flight; no notable difference from current master 6e35cb0

**Comment by santiago3dr** [comment]
flight with pixhawk mini (v3):
https://logs.px4.io/plot_app?log=9cdc2b7e-7248-4c59-8861-a9c7cb1813f1

compared to master under same conditions:
https://logs.px4.io/plot_app?log=f35edc0c-5af3-44e3-9e7b-caa6ee15cca1

good flight; no notable difference from current master 6e35cb0

**Comment by davids5** [comment]
@santiago3dr - Thank you!

@LorenzMeier - Please have a look at the PX4IO rcS start up, let me know if that change it is not OK.

Then let's merge this on Sept. 10th

**Comment by santiago3dr** [comment]
flight with pixhawk 3 (v4pro):
https://logs.px4.io/plot_app?log=22ba6954-4d08-48e7-b78d-871e285f4a4f

compared to master under same conditions:
https://logs.px4.io/plot_app?log=7583fa6b-2800-419f-a4fc-eaf4bed1d775

good flight; no notable difference from current master 6e35cb0

**Comment by davids5** [comment]
rebased on master cde35455603b8f697048af0e9368a6f01ab4ef18

**Comment by davids5** [comment]
@LorenzMeier this should come today.

**Comment by dagar** [comment]
Goodnight sweet prince.

**Pull request merged**